tee_supplicant/src/tee_supplicant.c

```diff
@@ -135,6 +135,66 @@ static int load_ta(uint32_t num_params, struct tee_param *params)
 	return TEEC_SUCCESS;
 }
 
+static int shm_alloc(const struct device *dev, uint32_t num_params,
+		     struct tee_param *params)
+{
+	void *addr;
+	size_t size;
+
+	if (num_params != 1) {
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	/*TODO: params[0].a is buffer type OPTEE_RPC_SHM_TYPE_*
+	 * params[0].c is alignment
+	*/
+	switch (params[0].attr & TEE_PARAM_ATTR_TYPE_MASK) {
+	case TEE_PARAM_ATTR_TYPE_VALUE_INPUT:
+	case TEE_PARAM_ATTR_TYPE_VALUE_OUTPUT:
+	case TEE_PARAM_ATTR_TYPE_VALUE_INOUT:
+		size = params[0].b;
+		break;
+	default:
+		return TEEC_ERROR_BAD_PARAMETERS;
+	}
+
+	/*TODO: check for TEE_GEN_CAP_REG_MEM */
+	addr = k_aligned_alloc(4096, size);
+	if (!addr) {
+		return TEEC_ERROR_OUT_OF_MEMORY;
+	}
+
+	params[0].c = (uint64_t)addr;
```

Comment:
At 149 line you noted that params[0].c is alignment, but then you put newly allocated buffer to params[0].c.
Is is some mistake?

Reply:
yes, mistake in comment, `params[0].c` is output parameter, `params[0].a` is shm allocation type (not used here). Fixed in [67a0bd9](https://github.com/dsemenets/zephyr-optee-client/pull/1/commits/67a0bd98cdc40cb4ed9ba99f5b88a4cdb4c2807d)